_posts/2018-06-29-node-red-dev-ci.md

```diff
@@ -0,0 +1,590 @@
+</div>
+
+Confirm these settings by clicking 'Done'.
+
+
+### Function block
+
+Double click the `function`-block. Again, a window will open!
+
+* Name: Retrieve data
+* Function:
+
+```javascript
+msg.payload = [
+  {id: 1, title: "Title 1"},
+  {id: 2, title: "Title 2"},
+  {id: 3, title: "Title 3"},
+  {id: 4, title: "Title 4"},
+  {id: 5, title: "Title 5"}
+];
+return msg;
+```
+
+NOTE:
+The information section on the right comes in handy while configuring the `function`-block
+
+> A JavaScript function block to run against the messages being received by the node.
+The messages are passed in as a JavaScript object called msg.
+By convention it will have a msg.payload property containing the body of the message.
+
+This explains why we are setting the `msg.payload` property.
+
+
+<div style="text-align: center;">
+  <img src="/img/node-red-dev-ci/config-function.png" width="300">
+</div>
+
+Again, confirm these settings by clicking 'done'.
+
+### Output
+
+Open the config for the `http-response`-output block.
+
+* Name: return all
+* Status Code: 200
+
+
+<div style="text-align: center;">
+  <img src="/img/node-red-dev-ci/config-output.png" width="300">
+</div>
+
+As you can see, the out put lets you choose the return status code. This can be used later if you would like to implement failed `REST`-calls for mocking purposes.
+Confirm these settings by clicking 'Done'.
+
+
+### Deploy
+
+On the top right of your screen, you have the deploy-dropdown. Clicking `deploy` will deploy this flow onto the running server, but you can also use the arrow to open up the menu and choose a more accurate scenario. Since this is our first deploy and we only have one flow, we don't have a choice.
+
+<div style="text-align: center;">
+  <img src="/img/node-red-dev-ci/deploy.png" width="300">
+</div>
+
+### Test setup
+
+To test our deployment and of course our flow, just navigate to http://localhost:1880/data in your browser or use curl to GET the data:
+
+<div style="text-align: center;">
+  <img src="/img/node-red-dev-ci/test-browser.png" width="300">
+</div>
+
+`$ curl -X GET http://localhost:1880/data`
+
+<div style="text-align: center;">
+  <img src="/img/node-red-dev-ci/test-curl.png" width="100%">
+</div>
+
+# Spicing things up
+
+Let's try and implement a `DELETE`.
+We would first drag and drop an `http`-input, a `function`-block and an `http-response`-output onto the flow chart.
+
+<div style="text-align: center;">
+  <img src="/img/node-red-dev-ci/delete.png" width="100%">
+</div>
+
+
+The input would look like:
+<div style="text-align: center;">
+  <img src="/img/node-red-dev-ci/delete-input.png" width="300">
+</div>
+
+Notice the `:id`. This way we tell Node-RED that this is a path parameter.
+
+The output would look like:
+<div style="text-align: center;">
+  <img src="/img/node-red-dev-ci/delete-output.png" width="300">
+</div>
+
+But what about the `function`-block ? We could come up with something like:
+
+```Javascript
```

Comment:
No code styling is being applied... maybe cause "Javascript" has a capital "J"? Try putting it lowercase.